spec/takeaway_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'takeaway'
+
+describe Takeaway do
+
+  subject{Takeaway.new(sms_sender)}
+  let(:sms_sender) { double(:sms_sender)}
+
+  it "responds to #menu"do
+    expect(subject).to respond_to(:menu)
+  end
+
+  describe "#read_menu" do
+
+    it "should have contents" do
+      expect(subject.menu.empty?).to be false
+    end
+
+  end
+
+  it "should respond to #place_order" do
+    expect(subject).to respond_to(:place_order).with(3).arguments
+  end
+
+  describe "#place order" do
+
+    before(:each) do
+      @order_list = {"Grilled banana" => 2}
+    end
+
+    it "should raise error if item not on menu" do
+      bad_order_list = {"Koala Pie" => 4}
+      expect{subject.place_order(bad_order_list, "£20", :phonenumber)}.to raise_error "Koala Pie not on menu."
```

Comment:
Line is too long. [110/80]<br>Space missing to the left of {.<br>Space missing inside {.<br>Space missing inside }.